src/Core/Utilities/CoreHelpers.cs

```diff
@@ -264,5 +265,37 @@ public static T Clone<T>(T obj)
         {
             return JsonConvert.DeserializeObject<T>(JsonConvert.SerializeObject(obj));
         }
+
+        public static string TextColorFromBgColor(string hexColor, int threshold = 166)
+        {
+            var bgColorObj = new ColorConverter().ConvertFromString(hexColor);
+            if (bgColorObj == null)
+            {
+                return "#ff000000";
+            }
+            var bgColor = (Color)bgColorObj;
+            var luminance = bgColor.R * 0.299 + bgColor.G * 0.587 + bgColor.B * 0.114;
+            return luminance > threshold ? "#ff000000" : "#ffffffff";
+        }
```

Comment:
I'd write this as:
```c#
if(new ColorConverter().ConvertFromString(hexColor) is Color bgColor)
{
    var luminance = bgColor.R * 0.299 + bgColor.G * 0.587 + bgColor.B * 0.114;
    return luminance > threshold ? "#ff000000" : "#ffffffff";
}

return "#ff000000";
```
Bear in mind that the conversion [can throw FormatException](https://learn.microsoft.com/en-us/dotnet/api/system.windows.media.colorconverter.convertfromstring?view=windowsdesktop-6.0#exceptions) but I'm hesitant to catch the exception on this method, I think it should be the responsability of the caller to provide a `hexColor` correctly formatted